content/en/docs/contributing/docs.md

```diff
@@ -16,10 +16,151 @@ You can add entire sections if you think a topic is missing or update existing o
+    |   |-- _index.md           # Community first page
+    |   |-- people.md           # List of people in the community
+    |   |-- publications.md     # List of Unikraft-related publications (theses, peer-reviewed publications, etc.)
+    |   `-- talks.md            # List of Unikraft-related talks and presentations
+    |
+    |-- docs/                   # Unikraft documentation, split by categories into subdirectories
+    |
+    |-- faq/                    # List of frequently asked questions
+    |
+    `-- releases/               # Unikraft releases summaries
+```
+
+### Testing Changes
+
+Before you submit your changes for review, please make sure you have tested it thoroughly.
+You can build and deploy the site locally, following [the instructions below](/contributing/docs/#building).
+You can also use the [Super Linter](https://github.com/github/super-linter), following the [instructions below](docs/contributing/docs/#using-the-linter).
+Please also do a spell check on your changes.
+Most text editors should be able to do that by themselves, [here's how you do it in `Vim`](https://thoughtbot.com/blog/vim-spell-checking).
+
+#### Building the website
+
+You can build the website inside a Docker environment or natively.
+Building inside a Docker environment is recommended, since you will work in the same environment as the deployed website.
+
+#### Building the website inside a Docker environment
+
+You can build the site inside a Docker environment.
+For this, you will need to install Docker-CE.
+You can do that by following the instructions [here](https://github.com/docker/docker-install), or by running:
+
+```console
+$ curl -fsSL https://get.docker.com/ | sh
+```
+
+**Note: Building the site inside a Docker environment is highly recommended, since you will work in the same environment as the deployed site, so there will be no errors regarding packages version or filesystem layout.**
+
+To then deploy the site, run:
+
+```console
+$ make container
+$ make devenv # This will take you inside the container
+
+# To be run inside the docker env
+$ npm install
+$ make serve
+```
+
+You may need admin privileges to run the first 2 commands.
+
+#### Building the website natively
+
+You can also build the site natively.
+For a native build, you need to install `hugo`.
+You can do it by following the instructions [here](https://gohugo.io/getting-started/installing/).
+
+Then, to build and deploy the site, run:
+
+```console
+$ hugo serve
+```
+
+**Note: This may lead to errors due to different `hugo` versions or other environment problems.
+The problems can be avoided by building the webside inside a Docker environment.
+If you choose to build the site natively, make sure you use [`Hugo v0.98.0`](https://github.com/gohugoio/hugo/releases/tag/v0.98.0).**
+
+After building and deploying the site, `hugo` provides instructions on accessing and using it:
+
+```text
+Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
```

Comment:
Could we provide a link here to that address, to be one-click away?

Reply:
I would not provide a link to a localhost address, people might get confused if it does not work.
The link will be available from the command output in their terminal.